<div class="lift:surround?with=math;at=math-content">
  <h2>Research</h2>
  <p>
  I study the processes, running inside a living cell.
  All vital intracellular processes are driven by proteins.
  Proteins are complex organic molecules, facilitating a wide variety of cellular functions.
  The following movie gives an idea about proteins in action.
  </p>
  
  <div align="center">
  	<iframe width="560" height="315" src="http://www.youtube.com/embed/9KTDz-ZisZ0" frameborder="0" allowfullscreen="true" ></iframe>
  	<br/>
  	<br/>
  </div>
  
  <h3>A living cell as a computer</h3>
  In mid 1940-s von-Neuman proposed that the chromosome of a living organism contains both the data for building a new organism and the description of the tools to be used. 
  
  In other words the gene expression mechanism of a cell along with the DNA molecule implement universal computational model.   
  
  <h3>How are intracellular processes coordinated?</h3>
  <p>
  Every vital process occurring inside a cell is driven by a protein or a number of proteins.
  A cell produces proteins in order to lunch some cellular process.
  A cellular process runs as long as corresponding proteins are present in a cell.
  A protein, like any other thermodynamical system, losses its original structure over time.
  The the loss of structure by a protein automatically leads to the loss of its cellular function and corresponding cellular process termination.  
    
  The cellular process terminat
  Proteins are able to affect on the production rate of other proteins.
    
  Proteins are made up of 20 different building blocks, called amino acids.
  Amino acids in a protein are arranged in chain, the majority of proteins are less than 300 amino acids in length.
  The amino acid chain entangles, forming unique 3-dimensional structure.
  
  
  The amino acid sequence of a protein is called his primary structure.
  
  The spatial spatial structure of a protein defines its function in cell.
  The spatial structure of a protein depends on its primary structure.
  So the amino acid sequence of a protein defines its features.
  
  Most proteins lose its initial structure in less than two days.
  A protein, like any other thermodynamical system, losses its original structure over time.
  Protein chains break down into shorter chains of amino acids.
  Short amino acid chains can form unpredictable spatial structures whose function may be harmful to the cell.
  The cell gets rid of the short amino acid chains, breaking them down into individual amino acids (the protein complex in charge is called Lysosome).
  </p>             
  
  
  <h3>How do proteins appear inside the cell when needed</h3>
  <p>
  The cell can not store all the necessary proteins, at least, because they get useless very fast.
  The cell can not wait for the necessary proteins to occur from the environment.
  The cell can not invent a new protein each time it is needed, because it is very hard.
  The problem of creating a protein by its predefined properties resembles NP-complete problem on the cellular level.
  The cell stores amino acid sequences of proteins in the structure of DNA molecule.
  DNA molecule is a chain, consisting of 4 building blocks(A,C,G,T) called nucleotides.
  Each of 20 amino acid is encoded by 3 nucleotides in DNA.  
  Protein complex called Ribosome assembles new proteins using the instructions encoded in DNA.
  Ribosome attaches amino acids one by one in the order they occur in the instructions.
  Proteins are able to turn up or down the synthesis of other proteins.
  </p>
  
  
  <h3>Why proteins as computer programs?</h3>
  <p>
  What happens inside the cell resembles computer software.
  von-Neuman's architecture of a computer.
  DNA along with the gene expression mechanism of the living cell implement universal computational model.
  DNA contains the description of the commands that are to be executed via gene expression mechanism.
  In computer software like in the cell any problem has multiple solutions. The optimal solution of the probem might be not easy to maintain or re-use.
  Living cells corresponds to computers, proteins correspond to computer programs, DNA corresponds to a hard drive and cytoplasm corresponds to RAM.
  Programs are stored on hard drives, proteins are stored in DNA.
  Prior to executing, a program is loaded to RAM from a hard drive, proteins are loaded from DNA to cytoplasm.
  When program completes the allocated memory is freed and ready to be used for running another program, proteins break down to independent amino acids, ready to be used in new protein.   
  Computer program is hard to create, but once it is created it can be copied and re-used almost for free.
  Instructions encoded in DNA correspond to the source code.
  Gene expression corresponds to computer program compilation and launching.
  </p>
     
  <h3>What are the properties of the programming language, that DNA programs are written in?</h3>
  <p>
  	While computer program is a sequence of actions to be performed by some computation unit, there is no evidence of any sequence of protein interactions defined in DNA.
  	DNA programs more resemble declarative programming: the presence of all involved proteins is enough for the process to run properly, since proteins automatically interact in right order.
  	Source code re-use practices of DNA programs resemble those of object oriented programming.
  	Most protein perform their function in complex with other proteins. Forming complexes of proteins corresponds to composition in OOP.
  	The domain structure of proteins correspond to inheritance in OOP because usually we are able to trace how the new genes were derived from the initial ones. 
  	Compartmetization of eucariota cell corresponds to muticore computers with own memory per core. The reactions in different compartments run independently.
  	Amino acids resemble basic computation units, which can be composed for obtaining more complex computation units.
  	The complexity of computational units is limited. The majority of protein domains is less than 200 amino acids in length. 
  	The length of the function body in imperative style programming language is about 20 lines, so to fit it on the computer display. 
  	DNA programming language is stable in a sense that replacement of a single amino acid by other one often preserves the function of protein.
  	In computer programming languages the mutation of a single symbol of the source code leads to compilation failures. 
  	DNA programming language is not very stable in a sense that insertion or deletion of a single symbol into the program's source code leads to the dramatics changes of protein function.
  	In computer programming languages insertion of a command often preserves the original function.  
  </p> 
     
  <h3>How protein activity is modeled</h3>
  <p>
  	Every molecule inside the cell, including proteins, corresponds to an active particle. 
  	Every active particle has a random field associated with it.
  	Every active particle has two attributes, defining the properties of the corresponding random field.
  	Interaction allows a particle to change the neighboring particles.
  	Tension measures the stability of a particle surrounded by the neighboring particles.
  	The interaction and the tension of a particle are calculated by its structure. 	
  	
  	Procedure Q implements the dynamics of a some configuration of particles.
  	Procedure Q calculates the properties of the random fields associated with the particles in the configuration and applies the calculated fields to the configuration itself.
  	In such a way, configuration defines not only the types of particles and their locations, but also defines the evolution of the configuration. 
  
  	Consider the finite set of bonds and two types of active particles: basic and composite.
  	Composite particles are obtained by binding together the simpler ones using the bonds.
  	Composite particles correspond to more sophisticated random fields.
  	Composite particle can be represented by a tree.
  	Composite particle can be represented by a sequence of symbols from finite alphabet.
  	Each particle can be encoded in the structure of other particle.
  </p>
  
  <h3>How does it help</h3>
  <p>
  </p>
  
  
  <span class="math">\omega \in \Omega</span>
  <i>Particlez</i> is the name of the open-source project I'm working on.
</div>
